ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -135,197 +98,271 @@ export default class ConfirmDecryptMessage extends Component {
+              onClick={scrollToBottom}
+              iconName={IconName.Arrow2Down}
+              ariaLabel={t('scrollDown')}
+              backgroundColor={BackgroundColor.primaryDefault}
+              borderRadius={BorderRadius.full}
+              color={IconColor.primaryInverse}
+              display={Display.Flex}
+              size={ButtonIconSize.Md}
+            />
+          )}
+          {hasDecrypted ? (
+            <div
+              className={classnames({
+                'request-decrypt-message__message-copy': true,
+                'request-decrypt-message__message-copy--pressed':
+                  copyToClipboardPressed,
+              })}
+              onClick={copyMessage}
+              onMouseDown={() => setCopyToClipboardPressed(true)}
+              onMouseUp={() => setCopyToClipboardPressed(false)}
+            >
+              <Tooltip
+                position="bottom"
+                title={
+                  hasCopied ? t('copiedExclamation') : t('copyToClipboard')
+                }
+                wrapperClassName="request-decrypt-message__message-copy-tooltip"
+                style={{ display: 'flex', alignItems: 'center' }}
+              >
+                <div className="request-decrypt-message__message-copy-text">
+                  {t('decryptCopy')}
+                </div>
+                <Icon
+                  name={hasCopied ? IconName.CopySuccess : IconName.Copy}
+                  color={IconColor.primaryDefault}
+                />
+              </Tooltip>
+            </div>
+          ) : (
+            <div />
+          )}
+        </div>
       </div>
-    );
-  };
-
-  renderFooter = () => {
-    const {
-      cancelDecryptMessage,
-      clearConfirmTransaction,
-      decryptMessage,
-      history,
-      mostRecentOverviewPage,
-      txData,
-    } = this.props;
-    const { trackEvent, t } = this.context;
-
-    return (
-      <PageContainerFooter
-        cancelText={t('cancel')}
-        submitText={t('decrypt')}
-        onCancel={async (event) => {
-          await cancelDecryptMessage(txData, event);
-          trackEvent({
-            category: MetaMetricsEventCategory.Messages,
-            event: 'Cancel',
-            properties: {
-              action: 'Decrypt Message Request',
-              legacy_event: true,
-            },
-          });
-          clearConfirmTransaction();
-          history.push(mostRecentOverviewPage);
-        }}
-        onSubmit={async (event) => {
-          await decryptMessage(txData, event);
-          trackEvent({
-            category: MetaMetricsEventCategory.Messages,
-            event: 'Confirm',
-            properties: {
-              action: 'Decrypt Message Request',
-              legacy_event: true,
-            },
-          });
-          clearConfirmTransaction();
-          history.push(mostRecentOverviewPage);
-        }}
+      <Footer
+        cancelDecryptMessage={cancelDecryptMessage}
```

Comment:
If we have to pass so many properties to the `Footer`, does that suggest we can encapsulate some of the logic inside the footer itself?

Plus do we want to decouple these callbacks from the outcome, so ideally we'd instead have `onCancelClick` and `onSubmitClick` etc?